pkg/users/account-details.js

```diff
@@ -355,3 +347,126 @@
+                .then(() => {
+                    setSelected([...selected, group]);
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const onSelectGroup = (event, selection) => {
+        if (selected.includes(selection)) {
+            removeGroup(selection);
+        } else {
+            addGroup(selection);
+        }
+    };
+
+    const chipGroupComponent = () => {
+        return (
+            <LabelGroup numLabels={10}>
+                {(selected || []).map((currentLabel, index) => {
+                    const optional = currentLabel !== primaryGroupName && superuser.allowed ? { onClose: () => removeGroup(currentLabel) } : {};
+
+                    return (
+                        <Label key={currentLabel}
+                               color={groups.find(group => group.name === currentLabel).isAdmin ? "gold" : "cyan"}
+                               {...optional}
+                        >
+                            {currentLabel}
+                        </Label>
+                    );
+                })}
+            </LabelGroup>
+        );
+    };
+
+    return (
+        <FormGroup
+            fieldId="account-groups"
+            helperText={
+                history.length > 0
+                    ? <HelperText className="pf-c-form__helper-text">
+                        <Flex>
+                            <HelperTextItem id="account-groups-helper" variant="warning">{_("The user must log out and log back in for the new configuration to take effect.")}</HelperTextItem>
+                            {history.length > 0 && <Button variant="link" id="group-undo-btn" isInline icon={<UndoIcon />} onClick={undoGroupChanges}>{_("Undo")}</Button>}
+                        </Flex>
+                    </HelperText>
+                    : ''
+            }
+            id="account-groups-form-group"
+            label={_("Groups")}
+            validated={history.length > 0 ? "warning" : "default"}
+        >
+            {superuser.allowed
+                ? <Select
+                   chipGroupComponent={chipGroupComponent()}
+                   isDisabled={!superuser.allowed}
+                   isOpen={isOpenGroup}
+                   onSelect={onSelectGroup}
+                   onToggle={setIsOpenGroup}
+                   selections={selected}
+                   toggleId="account-groups"
+                   variant={SelectVariant.typeaheadMulti}
+                >
+                    {groups.map((option, index) => (
+                        <SelectOption
+                            isDisabled={option.name == primaryGroupName}
+                            key={index}
+                            value={option.name}
+                        />
+                    ))}
+                </Select>
+                : chipGroupComponent()}
+        </FormGroup>
+    );
+};
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17996-20221212-195940-2c392d36-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)